huxley/www/js/components/AdvisorAssignmentsView.js

```diff
@@ -235,14 +235,19 @@ var AdvisorAssignmentsView = React.createClass({
     if (confirm) {
       AssignmentActions.updateAssignment(assignment.id, {
         rejected: true,
-      });
+      }, this._handleError);
     }
   },
 
   _handleSave: function(event) {
     var school = CurrentUserStore.getCurrentUser().school;
-    DelegateActions.updateDelegates(school.id, this.state.delegates);
-  }
+    DelegateActions.updateDelegates(school.id, this.state.delegates, this._handleError);
+  },
+
+  _handleError: function(response) {
+    window.alert("Something went wrong. Please refresh your page and try again.");
+    this.setState({loading: false});
```

Comment:
Do this before the alert, I think that alerts pause JS on the page (might be wrong though).